shopinvader_api_sale_loyalty/models/shopinvader_api_cart_service_helper.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, models
+
+from odoo.addons.base.models.res_partner import Partner as ResPartner
+from odoo.addons.sale.models.sale_order import SaleOrder
+from odoo.addons.shopinvader_api_cart.schemas import CartTransaction
+
+from ..schemas import LoyaltyRewardInput
+
+
+class ShopinvaderApiCartServiceHelper(models.AbstractModel):
+    _inherit = "shopinvader_api_cart.service.helper"
+
+    @api.model
+    def _sync_cart(
+        self,
+        partner: ResPartner,
+        cart: SaleOrder,
+        uuid: str,
+        transactions: list[CartTransaction],
+    ):
+        # TODO: will fail when several rewards are available
+        # TODO: was a skip_coupon_recompute in params...
+        cart = super()._sync_cart(partner, cart, uuid, transactions)
+        data = LoyaltyRewardInput()
+        self.env["shopinvader_api_loyalty.service.helper"]._recompute_coupon_lines(
+            cart, data
+        )
+        return cart
```

Comment:
In restAPI we recomputed coupon lines (to see if automatic rewards could be applied) each time an item was added to/deleted from the cart, except if a special context key was provided to skip the re-computation.
I'm not sure we should still do it here:
* We should extend Pydantic model `CardSyncInput` to take the `skip_recompute_coupons` context key.
* This re-computation will fail if an automatic loyalty program lets the user chose between several rewards. 

There exists another service `recompute_coupon_lines` that can be applied on a cart. If the convention is to call this service before validating the cart, we could improve the module to:
* For a given cart, get all automatic (i.e. without code) claimable rewards
* Specify, when recomputing coupon lines, which reward should be applied.